pennylane/vqe/vqe.py

```diff
@@ -129,6 +130,70 @@ def __str__(self):
 
         return "\n+ ".join(terms)
 
+    def __matmul__(self, H):
+
```

Comment:
Private methods don't need docstrings, but it may be helpful to include a short one for all these new methods so it's clearer what they do